File: JavaScript/3.2 - Readable tests/Calculator.js
// @ts-nocheck
let utils = require(`./utils`);

class Calculator {
  constructor() {}

  divide(n1) {
    let result = -9999;
    let separator = `,`;
    // if doesn't contain a comma call this
    if (!n1.includes(`,`)) {
      // separator is replaced with the result of the validSeparator call when passing n1
      separator = utils.validSeparator(n1); // slows down test
    }
    let numbers = n1.split(separator);

    for (let number of numbers) {
      if (!utils.isNumeric(number)) {
        return new Error(`Invalid numbers supplied`);
      }
    }

    if (numbers[1] == 0) {
      return new Error(
        `0 supplied as the divisor - unable to complete calculation`,
      );
    }

    result = utils.toNumber(numbers[0]) / utils.toNumber(numbers[1]);
    return result;
  }

  multiply(n1) {
    let result = -9999;
    let seperator = `,`;
    if (!n1.includes(`,`)) {
      // Seperator is replaced with the result of the validSeperaor call when passing N1
      seperator = utils.validSeparator(n1);
    }
    let numbers = n1.split(seperator);

    for (let number of numbers) {
      if (!utils.isNumeric(number)) {
        return new Error(`Invalid numbers supplied`);
      }
    }
    result = utils.toNumber(numbers[0]) * utils.toNumber(numbers[1]);
    return result;
  }

  subtract(n1) {
    let result = -9999;
    let separator = `,`;
    // if doesn't contain a comma call this
    if (!n1.includes(`,`)) {
      // separator is replaced with the result of the validSeparator call when passing n1
      separator = utils.validSeparator(n1); // slows down test
    }
    let numbers = n1.split(separator);
    for (let number of numbers) {
      if (!utils.isNumeric(number)) {
        return new Error(`Invalid numbers supplied`);
      }
    }

    result = utils.toNumber(numbers[0]) - utils.toNumber(numbers[1]);
    return result;
  }
}
module.exports = Calculator;

File: C#/3.2 - Readable tests/QACalculator/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QACalculator
{
    public class Calculator
    {
        public int Divide(String n1)
        {
            if (n1.Length > 0)
            {
                NewMethod1(n1);
            }
            else if(n1.Length == 0)
            {
                // if doesn't contain a comma call this
                if (!n1.Contains(","))
                    chat x = Utils.ValidSeperator(n1);    // slows down test

                foreach (String number in numbers)
                {
                    if (!Utils.IsNumeric(number))
                    {
                    }
                }

            }
            else
            {
                // something else
            }
            int result = -9999;
            char separator = ',';
            // if doesn't contain a comma call this
            if (!n1.Contains(","))
                separator = Utils.ValidSeperator(n1);    // slows down test
            String[] numbers = n1.Split(separator);
            foreach (String number in numbers)
            {
                if (!Utils.IsNumeric(number))
                {
                    return result;
                }
            }
            result = Utils.ToNumber(numbers[0]) / Utils.ToNumber(numbers[1]);
            return result;
        }

        private static void NewMethod1(string n1)
        {
            int result = -9999;
            char separator = ',';
            // if doesn't contain a comma call this
            if (!n1.Contains(","))
                separator = Utils.ValidSeperator(n1);    // slows down test
        }

        public int Multiply(String n1)
        {
            int result = -9999;
            String[] numbers = n1.Split(',');
            foreach (String number in numbers)
            {
                if (!Utils.IsNumeric(number))
                {  // check number is valid
                    return result;
                }
            }
            result = Utils.ToNumber(numbers[0]) * Utils.ToNumber(numbers[1]);
            return result;
        }

        public int Subtract(String n1)
        {
            int result = -9999;

            String[] numbers = n1.Split(',');
            foreach (String number in numbers)
            {
                if (!Utils.IsNumeric(number))
                {  // check number is valid
                    return result;
                }
            }
            result = Utils.ToNumber(numbers[0]) - Utils.ToNumber(numbers[1]);
            return result;
        }
    }
}
